### DISCUSSION TITLE
Confirmation pop-up before deleting anything

### DISCUSSION THREAD

**Discussion by jtwaleson** [darklang/dark#4554] [Feedback]
I clicked "delete" on a function and it went to the bin right away. Would be nice if there was a confirmation pop-up that explains what will happen (that can be overridden with shift, just like in windows explorer).

**Comment by jtwaleson** [comment]
Also the "+" symbol to restore an item is not great, should be :recycle: or something like this https://www.iconfinder.com/icons/1282797/backup_restore_time_machine_icon

**Comment by pbiggar** [comment]
> I clicked "delete" on a function and it went to the bin right away. Would be nice if there was a confirmation pop-up that explains what will happen (that can be overridden with shift, just like in windows explorer).

Our goal is for things to be instant, so it should be "it happens (safely) and you can undo it" rather than making you confirm. I think functions can't be deleted if they're in use.

But, looking at how we can make this feel safer, we should explain what happened to the thing you just deleted. (An alert box to say it can be undo from the trash, potentially with an undo button).

**Comment by jtwaleson** [comment]
Makes sense. Note that an HTTP endpoint can be called at any time from the
outside. As you’re changing the live app I see this as a disruptive
operation. In the cases where it’s not disruptive a pop-up afterwards seems
great!

**Reply by pbiggar** [reply]
Good point. The intent with HTTP handlers is that they "lock" as soon as they have some real traffic, but we haven't implemented that yet.

**Reply by sanderjson** [reply]
+1 for "lock" option, but always less popups
